### PR TITLE
MMDP-260 Enable searching of events

### PR DESCRIPTION
**Pull request by mutaimwiti**
**What does this PR do?**

Adds the ability to search events.

**Description of Task to be completed?**

As an administrator, I want to be able to search for a specific event on the 'Events' page by entering specific search criteria in a search box.

**How should this be manually tested?**

- Fetch and checkout to `ft-search-events-MMDP-260`.
- Run `yarn dev-server`.
- Open the frontend application and navigate to the events listing: `list-events`. You should see a list of all events or a prompt to add new events. If no events exist, add some with different titles.
- On the search input above the event, items type in a search string and click the search button. If your search matches the title of any of the events it will appear on a newly created listing that contains only the relevant results. If no event matches your search you will see no results.
- Erase all the search input and all events will appear.

**Any background context you want to provide?**

This solution accommodates/preserves pagination previously implemented on the page.

**What are the relevant Jira issues?**

[MMDP-260](https://viisaus.atlassian.net/browse/MMDP-260)

**Screenshots (if appropriate)**
#### Events listing
screenshot 2019-02-13 at 13 06 31

#### A search
screenshot 2019-02-13 at 13 07 40

#### Another search
screenshot 2019-02-13 at 13 08 16

#### Yet another search
screenshot 2019-02-13 at 13 08 04

**Questions:**
N/A

### CODE REVIEW

File: client/containers/events/eventsList.js [hunk 1]
```diff
@@ -9,27 +9,54 @@ import EmptyView from '../../components/common/InvalidPage';
...
+    this.fetchFirstPage();
+  };
+
+  handleSearchChange = (ev) => {
+    const { dispatch } = this.props;
+    const search = ev.target.value;
+
+    this.setState({ search });
+
+    if (search === '') {
```
Comment (shemogumbe) [inline client/containers/events/eventsList.js]: Always use falsiness test for empty strings i,.e if (!search) as the syntax is easier to undertand.

File: client/containers/events/eventsList.js [hunk 2]
```diff
@@ -9,27 +9,54 @@ import EmptyView from '../../components/common/InvalidPage';
...
+    search: '',
+  };
 
   componentDidMount() {
-    const { dispatch } = this.props;
-    dispatch(listEvents());
+    this.fetchFirstPage();
   }
 
+  fetchFirstPage = () => {
```
Comment (shemogumbe) [inline client/containers/events/eventsList.js]: We need to organize all pagination and pagination helpers into one common component, now seay the eventCrards will be displayed inside that paginated component that have everyone perform their own pagination on their component

File: client/containers/events/eventsList.js [hunk 3]
```diff
@@ -9,27 +9,54 @@ import EmptyView from '../../components/common/InvalidPage';
...
+    const search = ev.target.value;
+
+    this.setState({ search });
+
+    if (search === '') {
+      dispatch(listEvents({ page: 1, search }));
+    }
+  };
+
   handleNext = (e) => {
```
Comment (shemogumbe) [inline client/containers/events/eventsList.js]: This methods should form part of a common called Paginate

File: client/containers/events/eventsList.js [hunk 4]
```diff
@@ -9,27 +9,54 @@ import EmptyView from '../../components/common/InvalidPage';
...
     e.preventDefault();
     const { currentPage, dispatch } = this.props;
-    dispatch(listEvents(currentPage + 1));
+    const { search } = this.state;
+    dispatch(listEvents({ page: currentPage + 1, search }));
   };
 
   handlePrevious = (e) => {
```
Comment (shemogumbe) [inline client/containers/events/eventsList.js]: This should be part of a common called Paginate

File: client/containers/events/eventsList.js (hunk 2 above)
Comment (mutaimwiti) [reply client/containers/events/eventsList.js]: I will create a pagination component. I will introduce it in the groups search work.

File: client/containers/events/eventsList.js (hunk 3 above)
Comment (mutaimwiti) [reply client/containers/events/eventsList.js]: I will handle this in a pagination component.

File: client/containers/events/eventsList.js (hunk 4 above)
Comment (mutaimwiti) [reply client/containers/events/eventsList.js]: I will handle this in a pagination component.

File: client/containers/events/eventsList.js (hunk 2 above)
Comment (mutaimwiti) [reply client/containers/events/eventsList.js]: Pagination and search component  done

**Pull request merged**